### ISSUE TITLE
AffExpr iteration doesn't play nicely with mapreduce

### ISSUE THREAD

**Issue by joehuchette** [JuliaOpt/JuMP.jl#633]
As noticed on the [mailing list](), the current behavior for iteration over an AffExpr returning (coeff,var) doesn't work with the fallbacks for things like sum, mapreduce, etc.:
```jl
using JuMP
m = Model()
@defVar(m, x)
sum(2x) # => (2.0,x)
```
cc @IainNZ since I think you added this behavior?

**Comment by mlubin** [comment]
I think we should deprecate the current behavior. How about something like:
```
for (coeff,var) in linearterms(aff)
...
end
```

**Comment by IainNZ** [comment]
I like that proposal.
So we should remove the iteration protocol? What will `sum(2x)` do then?

**Comment by mlubin** [comment]
Should probably behave like iterating over a scalar, so ``sum(2x)`` will return ``2x``.

**Comment by joehuchette** [comment]
Closed by #636
